Gems/ROS2/Code/Include/ROS2/Manipulator/ManipulatorControllerComponent.h

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include <AzCore/Component/Component.h>
+#include <AzCore/Component/TickBus.h>
+#include <AzCore/Name/Name.h>
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+#include <ROS2/VehicleDynamics/DriveModels/PidConfiguration.h>
+
+#include <control_msgs/action/follow_joint_trajectory.hpp>
+#include "rclcpp_action/server.hpp"
+#include "rclcpp_action/rclcpp_action.hpp"
+
+
+namespace ROS2
+{
+    enum class GoalStatus
+    {
+        Pending,
+        Active,
+        Concluded
+    };
+
+    class FollowJointTrajectoryActionServer
+    {
+    public:
+        using GoalHandleFollowJointTrajectory = rclcpp_action::ServerGoalHandle<control_msgs::action::FollowJointTrajectory>;
```

Comment:
@adamdbrw Introducing an intermediate, exposed interface for effort/velocity seems like a very reasonable improvement to me. 

Nevertheless, revisiting the [overall proposal](https://github.com/o3de/o3de-ros2-gem/blob/manipulation_feature_design/docs/design/manipulation/analysis.md#considerations-for-implementation-within-the-o3de-simulator) and  [implementation roadmap](https://github.com/o3de/o3de-ros2-gem/blob/manipulation_feature_design/docs/design/manipulation/roadmap.md#motion-execution-and-simulation), how would you suggest to approach implementing what is called the `ROS2PoseTrajectory component` in the design docs instead?

Reply:
@lgleim I believe you are referring to this part:

> Create ROS2PoseTrajectoryComponent to receive the joint trajectory and execute the trajectory in simulation, where collisions, contacts, and dynamics are simulated.

This indeed suggests that the simulation should handle entire trajectory and I did see nor review it in the design doc, it seems that I have only seen an earlier revision, my apologies for not checking it thoroughly.

I do not believe this is the best approach since we are simulating/replacing what is actually a part of the robot stack logic to validate in simulation. This would be a bit similar to handling navigation goal inside the simulation, replacing what the navigation stack does and basing it on the ground truth. There, instead of handling high-level task that persist for a longer time (actions), the simulation handles basic commands such as velocity, steering and forces. Perhaps there is less logic happening in the realization of joint trajectory in some cases, but in other cases it might be subject to regulation, monitoring and interference with other robotic stack packages, and made to complied with safety monitors etc. Replacing the behavior with a simulated one would not be conductive to validation of such cases. 

Here is how would I suggest to approach pose trajectories for joints:
- have components for handling velocity and effort interfaces. For debug purposes, position interface can also be provided.
- have a joint trajectory controller running as a part of robotic stack (as it would be for on the real robot), but get the joint states from the simulation and realize its control interfaces (velocity/effort) through simulation.
- (optional) visualize trajectory controller state in simulation. Handling of the trajectory message by simulation could still be useful, for example to visualize it, allow the user to modify it etc..

Reply:
@michalpelka will re-cast the manipulation design docs PR to o3de-extras and we can submit some modifications to it together.

Reply:
@lgleim it is best to discuss it during a call, I would like to understand your perspective, perhaps simulating the entirety of trajectory action has more value than I am aware. Even so, it is still perfectly OK to start with the case (action server) that we might together consider to be useful to a smaller group of user, as long as the design supports introducing the other interface types quite well.

Reply:
@adamdbrw Would you mind verifying that https://github.com/o3de/sig-simulation/issues/54 covers the open aspects of this discussion, so that we can close this before moving the PR forward?